specs/select.spec.md

```diff
@@ -0,0 +1,156 @@
+# Select + Select-Option
+
+## Overview
+
+The `select` is a component that provides a list of options for the user to select from.
+
+### Background
+
+There is a standard select html control, `<select>` and `<option>`, but it doesn't support the full level of detail, styling, etc that most design systems need or require. For context please see [MDN Docs](https://developer.mozilla.org/en-US/docs/web/html/element/select).
+
+### Use Cases
+
+- *A customer using the component on a web page.*
+On a web page a customer is shopping for a new shirt in size medium. The customer opens the size `select` and sees a list of `<option>`s and selects size medium.
+
+### Features
+- form association
+- focus delegation
+
+### Features
+- Select one or multiple `<fast-option/>`(s) using the `multiple` attribute
+- When Dropdown does not have enough screen real estate to open below `<fast-select>` it will open above.
+
+### Prior Art/Examples
+- [FAST Select (React)](https://www.npmjs.com/package/@microsoft/fast-components-react-msft)
+- [Ant Design](https://ant.design/components/select/)
+- [Carbon Design](https://www.carbondesignsystem.com/components/select/code/)
+- [Atlassian UI](https://atlaskit.atlassian.com/packages/core/select)
+- [Lightning Design System](https://www.lightningdesignsystem.com/components/select/)
+- [W3 Example](https://www.w3.org/TR/wai-aria-practices-1.1/examples/listbox/listbox-collapsible.html)
+
+---
+
+### API
+
+*Component Name*:
+- `fast-select`
+
+*Attributes:*
+- `autocomplete` - Allows the developer to provide a hint on how to search the content within the `<fast-option>`(s)
+- `autofocus` - automatically focuses the control
+- `disabled` - disables the control
+- `length` - The number of `<fast-option>` elements in the `<fast-select>` element
+- `multiple` - Allows user to select more than one `<fast-option>`
+- `name` - the name of the control
+- `options` - Returns a `HTMLOptionsCollection` of the `<fast-option>` elements contained by the `<fast-select>` element
+- `required` - boolean value that sets the field as required
+- `selectedIndex` - The index of the first or last selected `<fast-option>` element, depending on the value of `multiple`
+- `selectedOptions` - An `HTMLCollection` of the selected `<fast-option>` elements
+- `size` - If the `<fast-select>` is shown with a scrollbar, this represents how many `<fast-option>`s are visible.
+- `value` - string value of the number field, can be an empty string
+
+*Events*
+- `change` - raise the change event for external parties to be informed of the `selectedIndex`'s value change.
```

Comment:
Do we need events around menu open/close?

Reply:
I'm not sure but open to suggestions here, the change event can be called when the menu opens and when the selection changes, (which would close the menu in single select mode) if we add an attribute `menu-open` change will return the context of the entire component knowing that the menu is open or close and what the selected index is.

Reply:
Also, should the open/close state of the menu be controllable by a developper via an attribute?  You will probably want an easy way to force the menu open during development in any case.

Reply:
Thats what I suggested above ;)